### DISCUSSION TITLE
`throw` a `validationError`

### DISCUSSION THREAD

**Discussion by moishinetzer** [airjp73/rvf#321] [Ideas]
It would be great, if instead of returning a deeply nested validation error, we would be able to `throw` the `validationError` just like you can `throw` a remix `redirect`. This would help with deeply nested service functions having to return the error, and having to return that from the `loader`, essentially making extracting out the logic an anti-pattern.

Are there any other patterns that others use to tackle this issue, or perhaps is there a "wrapper" or abstraction that can be applied to make this functionality work?

**Comment by airjp73** [comment]
I spent quite a bit of time in the beginning of this library trying to figure out how to achieve that. I'm right there with you haha. Unfortunately, I don't think it's possible without some kind of userland wrapper around your action.
